### ISSUE TITLE
lucene index plugin deployment

### ISSUE THREAD

**Issue by dejwsz** [Stratio/cassandra-lucene-index#44]
Just a question about deployment and use of this plugin. Suppose I have one cluster in data center 1 with cassandra with main db using vnodes. And now I would like to add another cluster in data center 2 and only there I would like to keep lucene indexes. Is it possible to separate things like that? I saw that there are some concepts with mixed deployments with cassandra where some nodes are used for main db, some for solr, spark etc.

**Comment by dejwsz** [comment]
From my tests I can see I have to configure plugin in both data centers (DC1 and DC2) if I have in between center replication enabled because after lucene index is created all node involved have to support it. This way the index data are stored everywhere the replication is configured (I'm talking about the community cassandra). So the only thing I can do to enforce index search only in the DC2 is to connect with one of the machines from DC2, set the consistency level as LOCAL_ONE or LOCAL_QUORUM and then execute select with lucene index where clause. Is that right approach? This way the index is consequently built in both data centers but the search burden is born only in DC2 where the vnodes are switched off. I understand that the index building in DC1 (with vnodes) is not such a heavy task when it is not used there actually? (besides it takes some disk space)

**Comment by dejwsz** [comment]
I'll be appreciate if someone can comment on it. Thanks.

**Comment by adelapena** [comment]
Hi,

You're right, the index is created in all the data centers and you can use LOCAL_* consistency to keep the search inside the connected data center. The tradeoff is the cost of writing the index in the data center that is not used for search. 

There are plans to add the ability to select the data centers in which the index creation is skipped. Ideally, this will be specified in the index creation statement. Alternatively, we could add a global configuration parameter to decide if a node is intended for search. Which approach do you think is more useful?

We will start working on it as soon as possible.

**Comment by dejwsz** [comment]
Hi,
thanks for response. Maybe it could be set while creating the index within options? Currently there is 'directory_path' setting, so maybe you can add something like 'target_data_center' where you can put data center name? How do you think?

**Comment by dejwsz** [comment]
And as I understand after such setting is available all queries will be directed (behind the scenes) only to the data centers which hold the given index related to the table/column family?

**Comment by adelapena** [comment]
No, you will still need to specify the LOCAL_* consistency. The changes will be done to skip the index writes in certain data centers.

**Comment by dejwsz** [comment]
OK

**Comment by dejwsz** [comment]
Are you able to state some approximate timeline for introducing this functionality? How do you think?

**Comment by adelapena** [comment]
Hi,

I have created a prototype of this new feature in the brach [feature/exclude_data_centers](https://github.com/Stratio/cassandra-lucene-index/tree/feature/exclude_data_centers).

I have added a new index property named 'excluded_data_centers' where you can specify the names of the index to be excluded from indexing:
```
CREATE CUSTOM INDEX demo_lucene 
ON  demo.users (lucene) 
USING 'com.stratio.cassandra.lucene.Index' WITH OPTIONS = {
'refresh_seconds':'1',  
'excluded_data_centers':'dc2,dc3',
 'schema':'...'};
```
The index will be created in these data centers, but all the write operations will be ignored. This way, ignored data centers will be able to route index searches but they wont consume resources writing.

**Comment by dejwsz** [comment]
Yes, I think definitely it is OK, many thanks!

**Comment by adelapena** [comment]
Great! Merged at master, the new feature will be available at 2.1.10.0 release.

**Comment by hbdestine** [comment]
Hi....Is it possible to keep the indexed folder entirely in a separate physical hard disk instead of different data centers.If it is possible then how to set the directory path(which is in a separate hard disk of a separate machine).Actually I was using the SimpleStrategy having only one dc.I am continuously getting the timeout error while reading the data.It becomes very hard for me to figure out what is the reason, because with the same query sometimes I m able to get the data but sometimes it throws request timeout error.So I was thinking of tuning.Thanks in advance.
